osu.Game.Rulesets.Osu/Skinning/Legacy/LegacySliderBall.cs

```diff
@@ -68,5 +76,25 @@ protected override void UpdateAfterChildren()
             layerNd.Rotation = -appliedRotation;
             layerSpec.Rotation = -appliedRotation;
         }
+
+        private void updateStateTransforms(DrawableHitObject obj, ArmedState _)
+        {
+            if (obj is not DrawableSlider)
+                return;
```

Comment:
I don't understand why this is required.

Reply:
Turns out it's not required because it's instant fade. The reason it was there was because it got called by slider ticks, tails, etc., and ended up stacking the fade out animation over and over, but since it's instant anyway I guess it doesn't result in any visual difference in this case.

Reply:
Hmm.... that doesn't sound great. Bets to leave this in place but also add a comment mentioning why.

I do wonder if we've implemented this incorrectly in other usages because that's a bit of a hidden detail that could potentially lead to performance issues.